WalletWasabi.Tests/UnitTests/BlockNotifierTests.cs

```diff
@@ -77,16 +79,41 @@ public async Task NotifyBlocksAsync()
 
 			// Assert that the blocks come in the right order
 			var height = 0;
-			void OnBlockInv(object s, Block b) => Assert.Equal(b.GetHash(), chain.GetBlock(height++).HashBlock);
+			string message = string.Empty;
+
+			void OnBlockInv(object blockNotifier, Block b)
+			{
+				uint256 h1 = b.GetHash();
+				uint256 h2 = chain.GetBlock(height + 1).HashBlock;
+
+				if (h1 != h2)
+				{
+					message = string.Format("height={0}, [h1] {1} != [h2] {2}", height, h1, h2);
+					cts.Cancel();
+					return;
+				}
+
+				height++;
+
+				if (height == BlockCount)
+				{
+					cts.Cancel();
+				}
+			}
+
 			notifier.OnBlock += OnBlockInv;
 
-			foreach (var n in Enumerable.Range(0, blockCount))
+			foreach (var n in Enumerable.Range(0, BlockCount))
 			{
 				await AddBlockAsync(chain);
 			}
 
 			notifier.TriggerRound();
-			await Task.Delay(TimeSpan.FromMilliseconds(100)); // give it time to process the blocks
+
+			// Wait at most 10 seconds or until cts is canceled
+			await WaitForCancelSignalAsync(TimeSpan.FromSeconds(10), cts.Token);
```

Comment:
This is not okay. There are some special cases in a few tests where there is a lot of IO operations and/or heavy CPU task like mining a block, detect the new block, generate the corresponding compact filter and get it from the coordinator component that can require 10 seconds or more but here there is no reason to wait up to 10 seconds when there is nothing to wait for really.

Reply:
10 seconds is an overkill, yes. It was supposed to be. But yeah, I can decrease it. I have just set it to 1500 milliseconds. That's still overkill. The thing is that it will take only 1500 ms when the test fails and unless something change it won't happen.

If you have a better magic constant in mind, please let me know.